plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/functions.tsx

```diff
@@ -0,0 +1,191 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { TableRow } from '@woocommerce/components/build-types/table/types';
+import { Icon, plugins } from '@wordpress/icons';
+import { gmdateI18n } from '@wordpress/date';
+import { Button } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { Subscription } from '../../types';
+import StatusPopover from './status-popover';
+import ActivationToggle from './activation-toggle';
+import ActionsDropdownMenu from './actions-dropdown-menu';
+
+function getStatus( subscription: Subscription ): {
+	text: string;
+	warning: boolean;
+	explanation?: string;
+} {
+	// TODO add statuses for subscriptions
+	if ( subscription.active ) {
+		return {
+			text: __( 'Active', 'woocommerce' ),
+			warning: false,
+		};
+	}
+
+	if ( subscription.product_key === '' ) {
+		return {
+			text: __( 'Not found', 'woocommerce' ),
+			warning: true,
+			explanation: __( 'This subscription is not found.', 'woocommerce' ),
+		};
+	}
+
+	if ( subscription.expired ) {
+		return {
+			text: __( 'Expired', 'woocommerce' ),
+			warning: true,
+			explanation: __(
+				'To receive updates and support for this extension, you need to purchase a new subscription or consolidate your extensions to one connected account by sharing or transferring this extension to this connected account.',
+				'woocommerce'
+			),
+		};
+	}
+
+	return {
+		text: __( 'Inactive', 'woocommerce' ),
+		warning: true,
+		explanation: __(
+			'To receive updates and support for this extension, you need to purchase a new subscription or consolidate your extensions to one connected account by sharing or transferring this extension to this connected account.',
+			'woocommerce'
+		),
+	};
+}
+
+function getVersion( subscription: Subscription ): string {
+	if ( subscription.local.version === subscription.version ) {
+		return subscription.local.version;
+	}
+
+	if ( subscription.local.version && subscription.version ) {
+		return subscription.local.version + ' > ' + subscription.version;
+	}
+
+	if ( subscription.version ) {
+		return subscription.version;
+	}
+
+	if ( subscription.local.version ) {
+		return subscription.local.version;
+	}
```

Comment:
When I look at the `Subscription` type, these properties always have a value. I guess it does no harm to be cautious. But doesn't this mean `subscription.local.version && subscription.version` is always true?

Reply:
`subscription.local.version` will be empty if the product isn't installed.
`subscription.version` can be empty if the API doesn't return this product.

This is why we have the two ifs bellow.